src/common/dtpthread.h

```diff
@@ -228,19 +251,25 @@ static inline int dt_pthread_rwlock_rdlock_with_caller(dt_pthread_rwlock_t *rwlo
 #define dt_pthread_rwlock_wrlock(A) dt_pthread_rwlock_wrlock_with_caller(A, __FILE__, __LINE__)
 static inline int dt_pthread_rwlock_wrlock_with_caller(dt_pthread_rwlock_t *rwlock, const char *file, int line)
 {
-  int res = pthread_rwlock_wrlock(&rwlock->lock);
-  rwlock->cnt ++;
+  const int res = pthread_rwlock_wrlock(&rwlock->lock);
+  assert(!res);
+  __sync_fetch_and_add(&(rwlock->cnt), 1);
```

Comment:
Ah, you are right, it's not an issue with normal builds, and debug builds should indeed be extra careful.